src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs

```diff
@@ -165,7 +166,10 @@ public static IReadOnlyList<PackageIdentity> GetRemovedPackages(LockFile origina
                 var closure = await project.GetProjectReferenceClosureAsync();
                 var files = closure.Select(reference => reference.PackageSpecPath).ToList();
 
-                var projectInfo = new BuildIntegratedProjectCacheEntry(project.JsonConfigPath, files);
+                var projectInfo = new BuildIntegratedProjectCacheEntry(
+                    project.JsonConfigPath,
+                    files,
+                    project.PackageSpec.RuntimeGraph.Supports.Keys.OrderBy(p => p, StringComparer.Ordinal).ToArray());
```

Comment:
The intent was if there was any change to the node we should re-restore. Changing the case seems like it.